devops/azure-pipelines/maven-build-test-job-template.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+  - name: 'jobName'
+    type: string
+  - name: 'serviceConnection'
+    type: string
+  - name: 'organization'
+    type: string
+  - name: 'testPath' # file path to test results
+    type: string
+jobs:
+  - job: ${{ parameters.jobName }}
+    steps:
+      - task: SonarCloudPrepare@1
+        condition: ne(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'Forge-SonarCloud'
+          organization: '1058-sophia-gavrila-forge'
+          scannerMode: 'Other'
+          extraProperties: |
+           sonar.branch.name=$(Build.SourceBranchName) 
+           sonar.projectKey=1058-Sophia-Gavrila-Forge_Forge-Backend
+           sonar.projectName=Forge-Backend
+      - task: SonarCloudPrepare@1
+        condition: eq(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'Forge-SonarCloud'
+          organization: '1058-sophia-gavrila-forge'
+          scannerMode: 'Other'
+          extraProperties: |
+           sonar.projectKey=1058-Sophia-Gavrila-Forge_Forge-Backend
+           sonar.projectName=Forge-Backend
+      - task: Maven@3
+        inputs:
+          mavenPomFile: 'pom.xml'
+          publishJUnitResults: true
+          testResultsFiles: '${{ parameters.testPath }}'
+          codeCoverageToolOption: 'JaCoCo'
+          codeCoverageRestoreOriginalPomXml: true
+          javaHomeOption: 'JDKVersion'
+          mavenVersionOption: 'Default'
+          mavenAuthenticateFeed: false
+          effectivePomSkip: false
+          sonarQubeRunAnalysis: true
+          sqMavenPluginVersionChoice: 'latest'
+      - task: PublishPipelineArtifact@1
+        inputs:
+          targetPath: '$(Pipeline.Workspace)/s/target/demo-0.0.1-SNAPSHOT.war'
```

Comment:
don't hard-code the ubuntu ms-hosted agent's `/s` directory structure, use a more appropriate predefined variable or just use a relative path.
also, the file name should probably be a parameter, no?

Reply:
`$(Build.Repository.LocalPath)` = `$(Pipeline.Workspace)/s/`
We forgot to change it to that, after being fully confused by path errors popping up like whack-a-mole.